README.md
# Dusk

Join us on our Discord at https://discord.gg/qzVpPgj

IMPORTANT: A severe bug with comeback gold and kill credits has been completely fixed as of 1.92b. Enjoy!

Play as one of 36 unique Heroes with powerful abilities! The same DotA map you know and love, but with custom Heroes and items.

Each Hero has been designed to feel powerful and impactful, with custom particles and sounds.

Upgrade late game items into their Exalted variants to crush your opposition, and buy into one of many new items, including the Holding Orb, Iris and Soul Vial.

Play the game mode as a 5v5/3v3 for the best experience. Please discuss any bugs, glitches or balance issues you find in the discussions.


CHANGELOG.md
# Changelog

## Alexander

* Base Strength increased from **23** to **32**.
* Strength gain increased from **2.0** to **3.4**.
* Base Intelligence reduced from **17** to **12**.
* Intelligence gain increased from **1.75** to **1.9**.
* Base Agility increased from **11** to **13**.
* **Radiant** is now **Raise the Shield**.
  * **Raise the Shield** creates a small area around Alexander, reducing the damage he and his allies within it receive.
* **Godfall** has been reworked.
  * Summons a rotating barrage of heavenly beams, dealing large damage over time in the area, and severely limiting the damage output of affected enemy units.
* **Elandra's Blessing** is now **Lightburst**.
  * Alexander unleashes a burst of light that heals him and damages enemies in a small area, dealing further damage over time.

## Astaroth

* Base Strength increased from **16** to **17**.
* Strength gain increased from **1.7** to **2.0**.
* **Deathmark** hits to activate increased from **5** to **6**.
* **Deathmark** now applies **20/30/40/50%** overall damage reduction to affected enemy Heroes for the debuff duration.

## Baal

* **Compression Sphere** cooldown reduced from **23s** to **14s**.
* **Compression Sphere** cast range reduced from **750** to **400**.
* **Spatial Rift** channel time rescaled from **0.75s** to **2.00/1.50/1.00/0.50s**.
* **Spatial Rift** cast range increased from **800/1000/1200/1400** to **1200**.
* **Spatial Rift** cooldown reduced from **17/16/15/14** to **9**.
* **Spatial Rift** mana cost reduced from **125** to **75**.
* **Spatial Rift** damage rescaled from **50/125/200/275** to **100/125/150/175**.

## Bahamut

* Hero changes.
* **Desolation** now reduces the main target's Magic Resistance by **15/20/25/30%**.
* **Desolation** cast range increased from **550** to **600**.
* **Desolation** cooldown increased from **16s** to **25/22/19/16s**.
* **Desolation** mana cost increased from **115/120/125/130** to **145**.
* **Desolation** damage per second increased from **10/20/30/40** to **15/25/35/45**.
* **Desolation** duration increased from **6.0s** to **8.0s**.
* **Desolation** radius increased from **400** to **550**.
* **Fulmination** has been reworked.
  * Now grants Bahamut **3** attacks that are infused with overwhelming energies, increasing their projectile speed, his attack range, changing the damage type to Magical, and dealing bonus damage.
* **Mega Flare** damage type changed from **Pure** to **Magical**.
* **Mega Flare** no longer pierces Spell Immunity.
* **Mega Flare** cast range reduced from **1500** to **975**.
* **Mega Flare** cast point reduced from **0.4s** to **0.3s**.
* **Mega Flare** cooldown reduced from **110** to **110/100/90**.
* **Mega Flare** mana cost increased from **230** to **350/400/450**.
* **Mega Flare** speed reduced from **450** to **315**.
* **Mega Flare** inner radius reduced from **325** to **275**.
* **Mega Flare** inner damage increased from **225/300/375** to **300/400/500**.
* **Mega Flare** outer damage increased from **175/225/275** to **200/250/300**.
* **Reckoning Aura** is now **Reckoning**.
  * Bahamut draws out the target's magical potential, taking it for himself for a short time - steals a portion of the target's Spell Amplification and slows them for a short time.

## Balthasar

* Hero changes.
* **Firebolt** cast range increased from **675** to **575**.
* **Firebolt** cooldown reduced from **14/13/12/11s** to **7s**.
* **Firebolt** now deals damage over time equal to **10/20/30/40** + **20/30/40/50%** of Balthasar's total attack damage.
* [TODO] Needs an ult change!

## Erra

* Hero changes.
* **Deathtouch** cast range increased from **150** to **175**.
* **Deathtouch** base damage over time increased to **80/160/240** damage per second.
* **Deathtouch** now deals bonus damage based on how much Health the enemy is missing, increasing by a maximum of **50%** when they hit **0%** Health.
* **To Dust** radius increased from **500** to **700** units.



game/scripts/vscripts/settings.lua
-- In this file you can set up all the properties and settings for your game mode.


ENABLE_HERO_RESPAWN = true              -- Should the heroes automatically respawn on a timer or stay dead until manually respawned
UNIVERSAL_SHOP_MODE = false             -- Should the main shop contain Secret Shop items as well as regular items
ALLOW_SAME_HERO_SELECTION = false       -- Should we let people select the same hero as each other

HERO_SELECTION_TIME = 60.0              -- How long should we let people select their hero?
PRE_GAME_TIME = 35.0                    -- How long after people select their heroes should the horn blow and the game start?
POST_GAME_TIME = 60.0                   -- How long should we let people look at the scoreboard before closing the server automatically?
TREE_REGROW_TIME = 180.0                 -- How long should it take individual trees to respawn after being cut down/destroyed?

GOLD_PER_TICK = 1                    -- How much gold should players get per tick?
GOLD_TICK_TIME = 0.50                      -- How long should we wait in seconds between gold ticks?

RECOMMENDED_BUILDS_DISABLED = false     -- Should we disable the recommened builds for heroes
CAMERA_DISTANCE_OVERRIDE = -1           -- How far out should we allow the camera to go?  Use -1 for the default (1134) while still allowing for panorama camera distance changes

MINIMAP_ICON_SIZE = 1                   -- What icon size should we use for our heroes?
MINIMAP_CREEP_ICON_SIZE = 1             -- What icon size should we use for creeps?
MINIMAP_RUNE_ICON_SIZE = 1              -- What icon size should we use for runes?

RUNE_SPAWN_TIME = 120                   -- How long in seconds should we wait between rune spawns?
CUSTOM_BUYBACK_COST_ENABLED = false     -- Should we use a custom buyback cost setting?
CUSTOM_BUYBACK_COOLDOWN_ENABLED = false  -- Should we use a custom buyback time?
BUYBACK_ENABLED = true            -- Should we allow people to buyback when they die?

DISABLE_FOG_OF_WAR_ENTIRELY = false     -- Should we disable fog of war entirely for both teams?
USE_UNSEEN_FOG_OF_WAR = false           -- Should we make unseen and fogged areas of the map completely black until uncovered by each team? 
                                            -- Note: DISABLE_FOG_OF_WAR_ENTIRELY must be false for USE_UNSEEN_FOG_OF_WAR to work
USE_STANDARD_DOTA_BOT_THINKING = false  -- Should we have bots act like they would in Dota? (This requires 3 lanes, normal items, etc)
USE_STANDARD_HERO_GOLD_BOUNTY = true    -- Should we give gold for hero kills the same as in Dota, or allow those values to be changed?

USE_CUSTOM_TOP_BAR_VALUES = false       -- Should we do customized top bar values or use the default kill count per team?
TOP_BAR_VISIBLE = true                  -- Should we display the top bar score/count at all?
SHOW_KILLS_ON_TOPBAR = true             -- Should we display kills only on the top bar? (No denies, suicides, kills by neutrals)  Requires USE_CUSTOM_TOP_BAR_VALUES

ENABLE_TOWER_BACKDOOR_PROTECTION = true-- Should we enable backdoor protection for our towers?
REMOVE_ILLUSIONS_ON_DEATH = false       -- Should we remove all illusions if the main hero dies?
DISABLE_GOLD_SOUNDS = false             -- Should we disable the gold sound when players get gold?

END_GAME_ON_KILLS = false                -- Should the game end after a certain number of kills?
KILLS_TO_END_GAME_FOR_TEAM = 50         -- How many kills for a team should signify an end of game?

USE_CUSTOM_HERO_LEVELS = false           -- Should we allow heroes to have custom levels?
MAX_LEVEL = 50                          -- What level should we let heroes get to?
USE_CUSTOM_XP_VALUES = false            -- Should we use custom XP values to level up heroes, or the default Dota numbers?

-- Fill this table up with the required XP per level if you want to change it
XP_PER_LEVEL_TABLE = {}
for i=1,MAX_LEVEL do
  XP_PER_LEVEL_TABLE[i] = (i-1) * 100
end

ENABLE_FIRST_BLOOD = true               -- Should we enable first blood for the first kill in this game?
HIDE_KILL_BANNERS = false               -- Should we hide the kill banners that show when a player is killed?
LOSE_GOLD_ON_DEATH = true               -- Should we have players lose the normal amount of dota gold on death?
SHOW_ONLY_PLAYER_INVENTORY = false      -- Should we only allow players to see their own inventory even when selecting other units?
DISABLE_STASH_PURCHASING = false        -- Should we prevent players from being able to buy items into their stash when not at a shop?
DISABLE_ANNOUNCER = false               -- Should we disable the announcer from working in the game?
FORCE_PICKED_HERO = nil                 -- What hero should we force all players to spawn as? (e.g. "npc_dota_hero_axe").  Use nil to allow players to pick their own hero.

FIXED_RESPAWN_TIME = -1                 -- What time should we use for a fixed respawn timer?  Use -1 to keep the default dota behavior.
FOUNTAIN_CONSTANT_MANA_REGEN = -1       -- What should we use for the constant fountain mana regen?  Use -1 to keep the default dota behavior.
FOUNTAIN_PERCENTAGE_MANA_REGEN = -1     -- What should we use for the percentage fountain mana regen?  Use -1 to keep the default dota behavior.
FOUNTAIN_PERCENTAGE_HEALTH_REGEN = -1   -- What should we use for the percentage fountain health regen?  Use -1 to keep the default dota behavior.
MAXIMUM_ATTACK_SPEED = 600              -- What should we use for the maximum attack speed?
MINIMUM_ATTACK_SPEED = 20               -- What should we use for the minimum attack speed?

GAME_END_DELAY = -1                     -- How long should we wait after the game winner is set to display the victory banner and End Screen?  Use -1 to keep the default (about 10 seconds)
VICTORY_MESSAGE_DURATION = 3            -- How long should we wait after the victory message displays to show the End Screen?  Use 
STARTING_GOLD = 500                     -- How much starting gold should we give to each player?
DISABLE_DAY_NIGHT_CYCLE = false         -- Should we disable the day night cycle from naturally occurring? (Manual adjustment still possible)
DISABLE_KILLING_SPREE_ANNOUNCER = false -- Shuold we disable the killing spree announcer?
DISABLE_STICKY_ITEM = false             -- Should we disable the sticky item button in the quick buy area?
SKIP_TEAM_SETUP = false                 -- Should we skip the team setup entirely?
ENABLE_AUTO_LAUNCH = true               -- Should we automatically have the game complete team setup after AUTO_LAUNCH_DELAY seconds?
AUTO_LAUNCH_DELAY = 60                  -- How long should the default team selection launch timer be?  The default for custom games is 30.  Setting to 0 will skip team selection.
LOCK_TEAM_SETUP = false                 -- Should we lock the teams initially?  Note that the host can still unlock the teams 


-- NOTE: You always need at least 2 non-bounty type runes to be able to spawn or your game will crash!
ENABLED_RUNES = {}                      -- Which runes should be enabled to spawn in our game mode?
ENABLED_RUNES[DOTA_RUNE_DOUBLEDAMAGE] = true
ENABLED_RUNES[DOTA_RUNE_HASTE] = true
ENABLED_RUNES[DOTA_RUNE_ILLUSION] = true
ENABLED_RUNES[DOTA_RUNE_INVISIBILITY] = true
ENABLED_RUNES[DOTA_RUNE_REGENERATION] = true
ENABLED_RUNES[DOTA_RUNE_BOUNTY] = true
ENABLED_RUNES[DOTA_RUNE_ARCANE] = true


MAX_NUMBER_OF_TEAMS = 2                -- How many potential teams can be in this game mode?
USE_CUSTOM_TEAM_COLORS = false         -- Should we use custom team colors?
USE_CUSTOM_TEAM_COLORS_FOR_PLAYERS = false          -- Should we use custom team colors to color the players/minimap?

TEAM_COLORS = {}                        -- If USE_CUSTOM_TEAM_COLORS is set, use these colors.
TEAM_COLORS[DOTA_TEAM_GOODGUYS] = { 61, 210, 150 }  --    Teal
TEAM_COLORS[DOTA_TEAM_BADGUYS]  = { 243, 201, 9 }   --    Yellow
TEAM_COLORS[DOTA_TEAM_CUSTOM_1] = { 140, 42, 244 }  --    Purple - for Nightmared units
TEAM_COLORS[DOTA_TEAM_CUSTOM_2] = { 255, 108, 0 }   --    Orange
TEAM_COLORS[DOTA_TEAM_CUSTOM_3] = { 52, 85, 255 }   --    Blue
TEAM_COLORS[DOTA_TEAM_CUSTOM_4] = { 101, 212, 19 }  --    Green
TEAM_COLORS[DOTA_TEAM_CUSTOM_5] = { 129, 83, 54 }   --    Brown
TEAM_COLORS[DOTA_TEAM_CUSTOM_6] = { 27, 192, 216 }  --    Cyan
TEAM_COLORS[DOTA_TEAM_CUSTOM_7] = { 199, 228, 13 }  --    Olive
TEAM_COLORS[DOTA_TEAM_CUSTOM_8] = { 140, 42, 244 }  --    Purple


USE_AUTOMATIC_PLAYERS_PER_TEAM = true   -- Should we set the number of players to 10 / MAX_NUMBER_OF_TEAMS?

CUSTOM_TEAM_PLAYER_COUNT = {}           -- If we're not automatically setting the number of players per team, use this table
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_GOODGUYS] = 5
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_BADGUYS]  = 5
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_CUSTOM_1] = 5
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_CUSTOM_2] = 0
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_CUSTOM_3] = 0
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_CUSTOM_4] = 0
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_CUSTOM_5] = 0
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_CUSTOM_6] = 0
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_CUSTOM_7] = 0
CUSTOM_TEAM_PLAYER_COUNT[DOTA_TEAM_CUSTOM_8] = 0
